src/components/API.js

```diff
@@ -0,0 +1,117 @@
+import { apiOptions } from "../utils/constants.js";
+
+export default class Api {
+  constructor() {
+    // constructor body
+    this._baseUrl = apiOptions.baseUrl;
+    this._headers = apiOptions.defaultHeaders;
+    this._userRoute = `${this._baseUrl}/users/me`;
+    this._cardRoute = `${this._baseUrl}/cards`;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    // if the server returns an error, reject the promise
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  async _request(url, options) {
```

Comment:
You created a custom fetch method, so the code that checks the server response and converts it from JSON is reused. Awesome!